aria/orchestrator/workflows/api/task.py

```diff
@@ -92,144 +93,133 @@ def __init__(self,
-            )
-
-        operations = interfaces.all()[0].operations.filter_by(name=name)
-        count = operations.count()
-        if count > 1:
-            raise exceptions.TaskException(
-                "More than one operation with the same name `{0}` found".format(name)
-            )
-        elif count == 0:
-            raise exceptions.TaskException(
-                "No operation with the name `{operation_name}` found".format(
-                    operation_name=name)
-            )
-
-        if not runs_on:
-            if edge == cls.SOURCE_OPERATION:
-                runs_on = model.Task.RUNS_ON_SOURCE
-            else:
-                runs_on = model.Task.RUNS_ON_TARGET
-
-        if runs_on == model.Task.RUNS_ON_SOURCE:
-            plugins = instance.source_node.plugins
-        else:
-            plugins = instance.target_node.plugins
-
-        return cls._instance(instance=instance,
-                             name=name,
-                             operation_template=operations[0],
-                             plugins=plugins or [],
-                             runs_on=runs_on,
-                             inputs=cls._merge_inputs(operations[0].inputs, inputs),
-                             *args,
-                             **kwargs)
 
-    @classmethod
-    def _instance(cls,
-                  instance,
-                  name,
-                  operation_template,
-                  inputs,
-                  plugins,
-                  runs_on,
-                  *args,
-                  **kwargs):
-        matching_plugins = [p for p in plugins if p['name'] == operation_template.plugin]
-        # All matching plugins should have identical package_name/package_version, so it's safe to
-        # take the first found.
-        plugin = matching_plugins[0] if matching_plugins else {}
-        return cls(actor=instance,
-                   name=name,
-                   implementation=operation_template.implementation,
-                   inputs=inputs,
-                   plugin=plugin,
-                   runs_on=runs_on,
-                   *args,
-                   **kwargs)
+        assert isinstance(relationship, models.Relationship)
+        assert runs_on in models.Task.RUNS_ON
+        return cls(
+            actor=relationship,
+            actor_type='relationship',
+            interface_name=interface_name,
+            operation_name=operation_name,
+            max_attempts=max_attempts,
+            retry_interval=retry_interval,
+            ignore_failure=ignore_failure,
+            inputs=inputs,
+            runs_on=runs_on)
+
+    @staticmethod
+    def _get_operation(interfaces, interface_name, operation_name):
+        interface = interfaces.get(interface_name)
+        if interface is not None:
+            return interface.operations.get(operation_name)
+        return None
+
+    @staticmethod
+    def _find_plugin(plugin_specification):
```

Comment:
I'm prepared to hear a better suggestion as to where to get the list of installed plugins from.